### PR TITLE
Introducing structured model outputs

### PR DESCRIPTION
**Pull request by laserkelvin**
This PR is semi-related to #104, allowing flexibility for wrappers and full implementations to potentially have a common output format that is also "backwards" compatible. It depends on `ModelOutput` from #315, so please review and merge that before this PR.

- Added a private attribute of the base `AbstractTask` class (which graph models inherit) called `__skip_output_heads__`, which `BaseTaskLitModules` will check this to determine if output heads are needed or not. If we do not use them, we do not initialize the `OutputHeads` to save on parameters/ensure we don't have unused parameters.
- I have refactored the `MACEWrapper` as an example of how to do this, and we should do so for other wrappers as well (e.g. `CHGNet`). I've also updated the `sphinx` documentation to detail how this can be used.
- I've ensured that the regular workflow (with PyG EGNN) isn't broken, and included unit tests for `MACEWrapper` to run the forward pass and an end-to-end with `ForceRegressionTask`.

### CODE REVIEW

**Comment by laserkelvin** [comment]
CI passed on relevant things

File: matsciml/models/base.py [hunk 1]
```diff
@@ -1871,10 +1922,38 @@ def forward(
...
+                encoder_outputs = self.encoder(batch)
+                if not isinstance(encoder_outputs, (Embeddings, dict, ModelOutput)):
+                    raise RuntimeError(
+                        f"Encoder model must emit a dict, `ModelOutput`, or `Embeddings` object. Got {encoder_outputs} instead."
+                    )
+                # sets the embeddings variable
+                if isinstance(encoder_outputs, Embeddings):
+                    embeddings = encoder_outputs
+                # for BYO output head cases
+                elif isinstance(encoder_outputs, ModelOutput):
```
Comment (melo-gonzo) [inline matsciml/models/base.py:1938]: A comment that may not be super important to consider right at this moment but should be noted - some workflows, such as serving models with OpenKIM's `kusp` to then run benchmarks, require that node energies are present. Some models (such as mace) can output these directly and they may be worth hanging onto as well.

**Review by melo-gonzo — APPROVED** [review]
Looks good overall! Will be a nice improvement to have. Left one general comment about downstream workflows.

File: matsciml/models/base.py (hunk 1 above)
Comment (laserkelvin) [reply matsciml/models/base.py:1938]: I think they're included in `ModelOutput` - in the wrapper output they're stashed as `node_energies`. That's what you mean right?

**Pull request merged**